In [2]:
library(ggplot2)
options(repr.plot.height=4,repr.plot.width=6)

library(dplyr)

Cargar los datos en un dataframe llamado: airbnb

In [3]:
airbnb<-read.csv('data//airbnb.csv',sep = ',', stringsAsFactors = T)

Mostrar las primeras 6 filas del dataframe

In [4]:
head(airbnb,6)

Zipcode Neighbourhood.Cleansed Property.Type Room.Type       Accommodates
1 28004   Universidad            Apartment     Private room    2           
2 28004   Universidad            Apartment     Entire home/apt 6           
3 28004   Universidad            Apartment     Entire home/apt 3           
4 28004   Universidad            Loft          Entire home/apt 3           
5 28015   Universidad            Apartment     Entire home/apt 5           
6 28004   Universidad            Apartment     Entire home/apt 2           
  Bathrooms Bedrooms Beds Bed.Type
1 2         1        1    Real Bed
2 1         3        5    Real Bed
3 1         2        2    Real Bed
4 2         1        1    Real Bed
5 1         1        1    Real Bed
6 1         0        1    Real Bed
  Amenities                                                                                                                                                                                                                                                                                                                                                       
1 TV,Wireless Internet,Kitchen,Pets allowed,Pets live on this property,Buzzer/wireless intercom,Heating,Suitable for events,Washer,First aid kit,Essentials,Lock on bedroom door,Iron                                                                                                                                                                             
2 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Indoor fireplace,Heating,Family/kid friendly,Washer,Dryer,Smoke detector,Carbon monoxide detector,Essentials,Shampoo                                                                                                                                                                                     
3 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Doorman,Heating,Family/kid friendly,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace                                                                                                                                                                                          
4 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Pets allowed,Breakfast,Elevator in building,Indoor fireplace,Buzzer/wireless intercom,Heating,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace,translation missing: en.hosting_amenity_49,translation missing: en.hosting_amenity_50                                          
5 TV,Wireless Internet,Air conditioning,Kitchen,Smoking allowed,Pets allowed,Elevator in building,Heating,Family/kid friendly,Washer,Essentials,Iron                                                                                                                                                                                                              
6 TV,Cable TV,Internet,Wireless Internet,Air conditioning,Wheelchair accessible,Kitchen,Doorman,Elevator in building,Buzzer/wireless intercom,Heating,Washer,Smoke detector,Carbon monoxide detector,First aid kit,Safety card,Fire extinguisher,Essentials,Shampoo,24-hour check-in,Hangers,Hair dryer,Iron,Laptop friendly workspace,Self Check-In,Doorman Entry
  Square.Feet Price Review.Scores.Rating
1  NA         35     NA                 
2  NA         92     96                 
3  NA         55     91                 
4  NA         60    100                 
5 538         75     90                 
6  NA         55     95

Renombrar las columnas de la siguiente forma:

| Nombre original | Nuevo nombre |
| - | - |
| Zipcode | CodigoPostal |
| Neighbourhood.Cleansed | Barrio |
| Property.Type	| TipoPropiedad |
| Room.Type | TipoAlquiler |
| Accommodates | MaxOcupantes |
| Bathrooms	| NumBanyos |
| Bedrooms	| NumDormitorios |
| Beds	|  NumCamas |
| Bed.Type	| TipoCama |
| Amenities | Comodidades |
| Square.Feet | PiesCuadrados |
| Price	| Precio |
| Review.Scores.Rating | Puntuacion |

In [5]:
newnames<-c("CodigoPostal","Barrio","TipoPropiedad","TipoAlquiler","MaxOcupantes","NumBanyos",
"NumDormitorios","NumCamas","TipoCama","Comodidades","PiesCuadrados","Precio","Puntuacion")
colnames(airbnb) <- newnames

Crea una nueva columna llamada MetrosCuadrados a partir de la columna PiesCuadrados.

Ayuda: 1 pie cuadrado son 0,092903 metros cuadrdados

In [6]:
airbnb

CodigoPostal Barrio           TipoPropiedad TipoAlquiler    MaxOcupantes
1     28004        Universidad      Apartment     Private room     2          
2     28004        Universidad      Apartment     Entire home/apt  6          
3     28004        Universidad      Apartment     Entire home/apt  3          
4     28004        Universidad      Loft          Entire home/apt  3          
5     28015        Universidad      Apartment     Entire home/apt  5          
6     28004        Universidad      Apartment     Entire home/apt  2          
7     28004        Universidad      House         Private room     7          
8     28004        Universidad      Apartment     Entire home/apt  3          
9     28015        Universidad      Apartment     Entire home/apt  2          
10    28004        Universidad      Apartment     Entire home/apt  4          
11    28004        Universidad      Apartment     Entire home/apt  4          
12    28004        Universidad      Apartment     Private room     2          
13    28013        Sol              Apartment     Private room     1          
14    28005        Sol              Apartment     Private room     2          
15    28013        Sol              Apartment     Entire home/apt  4          
16    28013        Sol              Condominium   Entire home/apt  6          
17    28013        Sol              Apartment     Entire home/apt  4          
18    28012        Sol              Apartment     Entire home/apt  6          
19    28013        Sol              Apartment     Entire home/apt  4          
20    28013        Sol              Apartment     Private room     2          
21    28013        Sol              Apartment     Entire home/apt  4          
22    28005        Sol              Apartment     Entire home/apt 10          
23    28013        Sol              Apartment     Entire home/apt  6          
24    28012        Sol              Apartment     Entire home/apt  4          
25    28012        Sol              Apartment     Entire home/apt  4          
26    28013        Sol              Apartment     Entire home/apt  4          
27    28014        Sol              Condominium   Entire home/apt  8          
28    28013        Sol              Apartment     Entire home/apt  6          
29    28012        Sol              Apartment     Private room     1          
30    28005        Sol              Apartment     Entire home/apt  8          
⋮     ⋮            ⋮                ⋮             ⋮               ⋮           
13178 28023        Valdemarín       House         Private room     1          
13179 28023        Aravaca          Condominium   Entire home/apt  3          
13180 28011        Puerta del Angel Apartment     Private room     2          
13181 28011        Puerta del Angel Other         Private room     2          
13182 28011        Puerta del Angel Apartment     Private room     3          
13183 28004        Universidad      Other         Entire home/apt  3          
13184 28004        Universidad      Apartment     Entire home/apt  3          
13185 28004        Universidad      Apartment     Entire home/apt  2          
13186 28015        Universidad      Apartment     Entire home/apt  4          
13187 28015        Universidad      Apartment     Entire home/apt  2          
13188 28015        Universidad      Apartment     Entire home/apt  5          
13189 28015        Universidad      Apartment     Private room     1          
13190 28015        Universidad      Apartment     Entire home/apt  4          
13191 28013        Universidad      Apartment     Entire home/apt  2          
13192 28004        Universidad      Apartment     Entire home/apt  4          
13193 28004        Universidad      Apartment     Entire home/apt 14          
13194 28004        Universidad      Apartment     Private room     1          
13195 28004        Universidad      Apartment     Entire home/apt  3          
13196 28004        Universidad      Apartment     Privat

In [7]:
#airbnb %>% select(PiesCuadrados)

In [8]:
#airbnb <- airbnb %>%
#  mutate(MetrosCuadrados = PiesCuadrados*0,3048)

airbnb$MetrosCuadrados <- airbnb$PiesCuadrados * 0.453592

In [9]:
head(airbnb,6)

CodigoPostal Barrio      TipoPropiedad TipoAlquiler    MaxOcupantes NumBanyos
1 28004        Universidad Apartment     Private room    2            2        
2 28004        Universidad Apartment     Entire home/apt 6            1        
3 28004        Universidad Apartment     Entire home/apt 3            1        
4 28004        Universidad Loft          Entire home/apt 3            2        
5 28015        Universidad Apartment     Entire home/apt 5            1        
6 28004        Universidad Apartment     Entire home/apt 2            1        
  NumDormitorios NumCamas TipoCama
1 1              1        Real Bed
2 3              5        Real Bed
3 2              2        Real Bed
4 1              1        Real Bed
5 1              1        Real Bed
6 0              1        Real Bed
  Comodidades                                                                                                                                                                                                                                                                                                                                                     
1 TV,Wireless Internet,Kitchen,Pets allowed,Pets live on this property,Buzzer/wireless intercom,Heating,Suitable for events,Washer,First aid kit,Essentials,Lock on bedroom door,Iron                                                                                                                                                                             
2 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Indoor fireplace,Heating,Family/kid friendly,Washer,Dryer,Smoke detector,Carbon monoxide detector,Essentials,Shampoo                                                                                                                                                                                     
3 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Doorman,Heating,Family/kid friendly,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace                                                                                                                                                                                          
4 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Pets allowed,Breakfast,Elevator in building,Indoor fireplace,Buzzer/wireless intercom,Heating,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace,translation missing: en.hosting_amenity_49,translation missing: en.hosting_amenity_50                                          
5 TV,Wireless Internet,Air conditioning,Kitchen,Smoking allowed,Pets allowed,Elevator in building,Heating,Family/kid friendly,Washer,Essentials,Iron                                                                                                                                                                                                              
6 TV,Cable TV,Internet,Wireless Internet,Air conditioning,Wheelchair accessible,Kitchen,Doorman,Elevator in building,Buzzer/wireless intercom,Heating,Washer,Smoke detector,Carbon monoxide detector,First aid kit,Safety card,Fire extinguisher,Essentials,Shampoo,24-hour check-in,Hangers,Hair dryer,Iron,Laptop friendly workspace,Self Check-In,Doorman Entry
  PiesCuadrados Precio Puntuacion MetrosCuadrados
1  NA           35      NA              NA       
2  NA           92      96              NA       
3  NA           55      91              NA       
4  NA           60     100              NA       
5 538           75      90        244.0325       
6  NA           55      95              NA

Miremos el código postal. Es una variable con entradas erroneas.
Hay valores como '', '-'  y '28' que deberían ser considerados como NA.
Así mismo también debería ser NA todos los que no compiencen por 28, ya que estamos con códigos postales de Madrid

El código postal 28002, 28004 y 28051 tienen entradas repetidas. 
Por ejemplo las entradas 28002\n20882 deberían ir dnetro de 28002

El codigo 2804 debería ser 28004, 2805 deberia ser 28005 y 2815 juncto con 2815 debería ser 28015

Limpia los datos de la columna Codigo Postal

In [84]:
?substring

In [87]:
grep("28002\n20882", pattern="\n") == 1

[1] TRUE

In [68]:
#table(substring(airbnb$CodigoPostal,1,2))
#table(substring(airbnb$CodigoPostal,1,2) == 28)
filter(airbnb, substring(airbnb$CodigoPostal,1,2) == 28)

CodigoPostal Barrio           TipoPropiedad TipoAlquiler    MaxOcupantes
1     28004        Universidad      Apartment     Private room     2          
2     28004        Universidad      Apartment     Entire home/apt  6          
3     28004        Universidad      Apartment     Entire home/apt  3          
4     28004        Universidad      Loft          Entire home/apt  3          
5     28015        Universidad      Apartment     Entire home/apt  5          
6     28004        Universidad      Apartment     Entire home/apt  2          
7     28004        Universidad      House         Private room     7          
8     28004        Universidad      Apartment     Entire home/apt  3          
9     28015        Universidad      Apartment     Entire home/apt  2          
10    28004        Universidad      Apartment     Entire home/apt  4          
11    28004        Universidad      Apartment     Entire home/apt  4          
12    28004        Universidad      Apartment     Private room     2          
13    28013        Sol              Apartment     Private room     1          
14    28005        Sol              Apartment     Private room     2          
15    28013        Sol              Apartment     Entire home/apt  4          
16    28013        Sol              Condominium   Entire home/apt  6          
17    28013        Sol              Apartment     Entire home/apt  4          
18    28012        Sol              Apartment     Entire home/apt  6          
19    28013        Sol              Apartment     Entire home/apt  4          
20    28013        Sol              Apartment     Private room     2          
21    28013        Sol              Apartment     Entire home/apt  4          
22    28005        Sol              Apartment     Entire home/apt 10          
23    28013        Sol              Apartment     Entire home/apt  6          
24    28012        Sol              Apartment     Entire home/apt  4          
25    28012        Sol              Apartment     Entire home/apt  4          
26    28013        Sol              Apartment     Entire home/apt  4          
27    28014        Sol              Condominium   Entire home/apt  8          
28    28013        Sol              Apartment     Entire home/apt  6          
29    28012        Sol              Apartment     Private room     1          
30    28005        Sol              Apartment     Entire home/apt  8          
⋮     ⋮            ⋮                ⋮             ⋮               ⋮           
12731 28023        Valdemarín       House         Private room     1          
12732 28023        Aravaca          Condominium   Entire home/apt  3          
12733 28011        Puerta del Angel Apartment     Private room     2          
12734 28011        Puerta del Angel Other         Private room     2          
12735 28011        Puerta del Angel Apartment     Private room     3          
12736 28004        Universidad      Other         Entire home/apt  3          
12737 28004        Universidad      Apartment     Entire home/apt  3          
12738 28004        Universidad      Apartment     Entire home/apt  2          
12739 28015        Universidad      Apartment     Entire home/apt  4          
12740 28015        Universidad      Apartment     Entire home/apt  2          
12741 28015        Universidad      Apartment     Entire home/apt  5          
12742 28015        Universidad      Apartment     Private room     1          
12743 28015        Universidad      Apartment     Entire home/apt  4          
12744 28013        Universidad      Apartment     Entire home/apt  2          
12745 28004        Universidad      Apartment     Entire home/apt  4          
12746 28004        Universidad      Apartment     Entire home/apt 14          
12747 28004        Universidad      Apartment     Private room     1          
12748 28004        Universidad      Apartment     Entire home/apt  3          
12749 28004        Universidad      Apartment     Privat

In [16]:
count(filter(airbnb, CodigoPostal == ""))

n  
1 439

In [18]:
filter(airbnb, CodigoPostal == "-")

CodigoPostal Barrio    TipoPropiedad TipoAlquiler    MaxOcupantes NumBanyos
1 -            Recoletos House         Entire home/apt 10           6        
  NumDormitorios NumCamas TipoCama Comodidades PiesCuadrados Precio Puntuacion
1 4              9        Real Bed             NA            NA     NA        
  MetrosCuadrados
1 NA

In [19]:
count(filter(airbnb, CodigoPostal == "28"))

n
1 1

In [92]:
filter(airbnb, CodigoPostal == "28002\n28002")

CodigoPostal Barrio      TipoPropiedad TipoAlquiler MaxOcupantes NumBanyos
1 28002\n28002 Prosperidad Apartment     Private room 2            1        
  NumDormitorios NumCamas TipoCama
1 1              1        Real Bed
  Comodidades                                                                                                                                                                                                                                                                                                          
1 TV,Internet,Wireless Internet,Air conditioning,Wheelchair accessible,Kitchen,Free parking on premises,Doorman,Elevator in building,Indoor fireplace,Buzzer/wireless intercom,Heating,Washer,First aid kit,Fire extinguisher,Essentials,Shampoo,Lock on bedroom door,Hangers,Hair dryer,Iron,Laptop friendly workspace
  PiesCuadrados Precio Puntuacion MetrosCuadrados
1 NA            70     100        NA

In [93]:
count(filter(airbnb, CodigoPostal == "28002"))

n  
1 170

In [98]:
count(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28002'))

n  
1 171

In [10]:
airbnb$CodigoPostal[airbnb$CodigoPostal==''] <- NA

In [ ]:
#Miremos el código postal. Es una variable con entradas erroneas. 
#Hay valores como '', '-' y '28' que deberían ser considerados como NA. 
#Así mismo también debería ser NA todos los que no compiencen por 28, 
#ya que estamos con códigos postales de Madrid

#El código postal 28002, 28004 y 28051 tienen entradas repetidas. 
#Por ejemplo las entradas 28002\n20882 deberían ir dnetro de 28002

#El codigo 2804 debería ser 28004, 2805 deberia ser 28005 y 2815 juncto con 2815 debería ser 28015

#Limpia los datos de la columna Codigo Postal

Una vez limpios los datos ¿Cuales son los códigos postales que tenemos? 

In [60]:
print("28002\n20882")

[1] "28002\n20882"


¿Cuales son los 5 códigos postales con más entradas? ¿Y con menos? ¿Cuantas entradas tienen?

¿Cuales son los barrios que hay en el código postal 28012?

¿Cuantas entradas hay en cada uno de esos barrios para el codigo postal 28012?

¿Cuantos barrios hay en todo el dataset airbnb? ¿Cuales son?

¿Cuales son los 5 barrios que tienen mayor número entradas?

¿Cuantos Tipos de Alquiler diferentes hay? ¿Cuales son? ¿Cuantas entradas en el dataframe hay por cada tipo?

## Second section

Muestra el diagrama de cajas del precio para cada uno de los diferentes Tipos de Alquiler

Cual es el precio medio de alquiler de cada uno, la diferencia que hay ¿es estadísticamente significativa?
¿Con que test lo comprobarías?

Filtra el dataframe cuyos tipo de alquiler sea  'Entire home/apt' y guardalo en un dataframe llamado 
*airbnb_entire*.
Estas serán las entradas que tienen un alquiler del piso completo.

¿Cuales son los 5 barrios que tienen un mayor número de apartamentos enteros en alquiler?
Nota: Mirar solo en airbnb_entire. A partir de este punto y hasta que se diga lo contrario partiremos de airbnb_entire.

¿Cuales son los 5 barrios que tienen un mayor precio medio de alquiler para apartamentos enteros (es decir, del dataframe airbnb_entire)?

¿Cual es su precio medio?

Ayuda: Usa la función aggregate `aggregate(.~colname,df,mean,na.rm=TRUE)`

¿Cuantos apartamentos hay en cada uno de esos barrios?

Mostrar una dataframe con el nombre del barrio, el precio y el número de entradas.

Ayuda: Podeis crear un nuevo dataframe con las columnas "Barrio" y "Freq" que contenga el número de entradas en cada barrio y hacer un merge con el dataframe del punto anterior.

Partiendo del dataframe anterior, muestra los 5 barrios con mayor precio, pero que tengan más de 100 entradas de alquiler.

Dibuja el diagrama de densidad de distribución de los diferentes precios. Serían 5 gráficas, una por cada barrio.

Calcula el tamaño medio, en metros cuadrados, para los 5 barrios anteriores y muestralo en el mismo dataframe junto con el precio y número de entradas

Dibuja el diagrama de densidad de distribución de los diferentes tamaños de apartamentos. Serían 5 gráficas, una por cada barrio.

Esta claro que las medias de metros cuadrados de cada uno de estos 5 barrios parecen ser diferentes, pero ¿son estadísticamente diferentes?
¿Que test habría que usar para comprobarlo?

Para únicamente los pisos de alquiler en el barrio de Sol:

``barrio_sol<-subset(airbnb_entire,Barrio=="Sol")``

Calcular un modelo lineal que combine alguna de estas variables:
* NumBanyos
* NumDormitorios
* MaxOcupantes
* MetrosCuadrados

Primero calculamos la correlación para ver como se relacionan estas variables entre sí.

Se observa que la correlación entre el número de dormitorios y los metros cuadrados es sorprendentemente baja.
¿Son de fiar esos números?

Mediante un histograma o curvas de densidad podemos descartar números que notienen sentido en el dataframe barrio_sol,
para tener una matriz de correlación que tenga mayor sentido.


Una vez que hayamos filtrado los datos correspondientes calcular el valor o la combinación de valores que mejor nos permite obtener el precio de un inmueble.

¿Que variable es más fiable para conocer el precio de un inmueble, el número de habitaciones o los metros cuadrados?

Responde con su correspondiente margen de error del 95%, ¿cuantos euros incrementa el precio del alquiler por cada metro cuadrado extra del piso?

Responde con su correspondiente margen de error del 95%, ¿cuantos euros incrementa el precio del alquiler por cada habitación?

¿Cual es la probabilidad de encontrar, en el barrio de Sol, un apartamento en alquiler con 3 dormitorios?
¿Cual es el intervalo de confianza de esa probabilidad?